kernel/init.c

```diff
@@ -465,6 +466,12 @@ FUNC_NORETURN void z_cstart(void)
 
 	/* perform basic hardware initialization */
 	z_sys_init_run_level(_SYS_INIT_LEVEL_PRE_KERNEL_1);
+
+#if defined(CONFIG_GDBSTUB) && \
+	defined(CONFIG_GDBSTUB_SERIAL_BACKEND)
+	gdb_start();
```

Comment:
instead of hacking this into here, just define a SYS_INIT() for it.

Reply:
ok, done. I've thought about it before but was not sure if that should be treated this way.